src/firefly/js/util/Interp/LinearInterpolator.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * Lijun Zhang
+ * 05/01/2017
+ * The LinearInterpolator interpolates the one dimensional data points to desired length based on Linear Interpolation
+ * algorithm.  The input data array has to be monotonically ascending or descending.  The minimum length of the input array
+ * is 2.  For any give coordinate (or index of the output array), the LinearInterpolator finds the value of the output array
+ * at that index (or coordinate) based on the input data.
+ */
+
+/**
+ * @Desc This method is displaying error message when there are exceptions.
+ * @param message
+ * @constructor
+ */
+function UserException(message) {
+    this.message = message;
+    this.name = 'UserException';
+}
+
+/**
+ * LinearInterpolator class
+ */
+export class LinearInterpolator {
+    constructor(x, y, allowExtrapolation) {
```

Comment:
I don't think this needs to be a class.  I should be a function closure that returns a function that will do that work (`getInterpolatedValue`). if there is an invariant it could return a null.  The other function could be in the closure since they are private anyway.